packages/millicast-sdk/src/utils/Diagnostics.js

```diff
@@ -39,9 +40,14 @@ const getDiagnostics = (statsCount = MAX_STATS_HISTORY_SIZE) => {
   diagnostics.version = version
   diagnostics.timestamp = Date.now()
   diagnostics.userAgent = window?.navigator?.userAgent || 'No user agent available'
+  diagnostics.stats = [...tempStats]
 
-  if (diagnostics.stats.length && Number.isInteger(statsCount) && statsCount >= 0 && statsCount < MAX_STATS_HISTORY_SIZE) {
-    diagnostics.stats = diagnostics.stats.slice(-statsCount)
+  if (tempStats.length && Number.isInteger(statsCount) && statsCount >= 0 && statsCount < MAX_STATS_HISTORY_SIZE) {
```

Comment:
If the condition is failed, we still return the result? The stats could still hold the old records.

Reply:
Would it be OK to reset the stats?

Reply:
No, we just provide the latest stats, it's up to the user if they want to save them or not.

Reply:
we have checked the statsCount, if it's invalid it will be MAX_STATS_HISTORY_SIZE, isn't it? And I think we should return the latest stats which meet the required statsCount, that's it.